File: unit-3/sprint-3/day-4/assignments/StopWatch/app/src/main/java/com/example/stopwatch/MainActivity.java
package com.example.stopwatch;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.SystemClock;
import android.view.View;
import android.widget.Chronometer;

public class MainActivity extends AppCompatActivity {
    private Chronometer stopWatch;
    private boolean isRunning;
    private long PauseMemory;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initview();
        stopWatch.setFormat("%s");
        stopWatch.setBase(SystemClock.elapsedRealtime());
//        stopWatch.setOnChronometerTickListener(chronometer -> {
//            if ()
//        });
    }

    private void initview() {
        stopWatch = findViewById(R.id.StopWatch);
    }

    public void startStopWatch(View view){
        if (!isRunning) {
            stopWatch.setBase(SystemClock.elapsedRealtime()-PauseMemory);
            stopWatch.start();
            isRunning = true;
        }

    }
    public void PauseStopWatch(View view){
        if (isRunning) {
            stopWatch.stop();
            PauseMemory = SystemClock.elapsedRealtime() - stopWatch.getBase();
            isRunning = false;
        }
    }
    public void ResetStopWatch(View view){
        stopWatch.setBase(SystemClock.elapsedRealtime());
        PauseMemory =0;

    }
}

File: unit-3/sprint-4/evaluation/App1/app/src/main/java/com/example/app1/MainActivity.kt
package com.example.app1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    private lateinit var resultList: List<Result>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnSearch.setOnClickListener {
            callAPi()
        }
    }

    private fun callAPi() {
        val apiClient  =  Network.getInstance().create(ApiClient::class.java)
        apiClient.getSong(etSearchQuery.text.toString()).enqueue(object : Callback<ResponseModel>{
            override fun onResponse(call: Call<ResponseModel>, response: Response<ResponseModel>) {
              response?.run {
                    resultList = body()?.results!!
                  setRecyclerview()

                }
            }

            override fun onFailure(call: Call<ResponseModel>, t: Throwable) {
                Toast.makeText(this@MainActivity,"Data Fetching failed",Toast.LENGTH_SHORT).show()
            }

        })
    }

    private fun setRecyclerview() {
        val songAdapter = SongAdapter(resultList)
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerview.adapter = songAdapter
        recyclerview.layoutManager = linearLayoutManager
    }
}